Enclave Prototype

itstauq opened the pull request:
## Description

Todo:
- [ ] Copy reviews (Review all the text that a user would see; for Product team?)
- [ ] Handle backward-compatible objects migrations

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

itstauq committed:
Add authorization checks for uploading assets to the enclave

itstauq committed:
Linter fix

itstauq committed:
Merge pull request #8953 from OpenMined/tauquir/enclave-users

Enclave authentication and authorization management

itstauq committed:
Merge branch 'dev' into tauquir/enclave-prototype

itstauq committed:
Remove left-over comments

itstauq committed:
Prevent EnclaveInstance from contacting enclave when not creating.

itstauq committed:
Fix error not being displayed properly; add navigation links in notebook

itstauq committed:
Merge branch 'dev' into tauquir/enclave-prototype

itstauq committed:
Add async job flow for EnclaveService.execute_code

itstauq committed:
Split EnclaveService into DomainEnclaveService and EnclaveService

BrendanSchell committed:
add distributed project id that's shared with client projects

BrendanSchell committed:
change deployment policy to runtime policy

itstauq committed:
Merge pull request #8979 from OpenMined/tauquir/split-enclave-service

Split EnclaveService into DomainEnclaveService and EnclaveService

BrendanSchell committed:
Merge pull request #8980 from OpenMined/bschell/make-projects-share-id-in-network-nodes

Add distributed project ID that's shared with client projects

itstauq committed:
Fix non-blocking code call by using client instead of node service

snwagh committed:
Merge pull request #8978 from OpenMined/tauquir/enclave-execute-code-refactor

Add async job flow for code execution in Enclaves

itstauq committed:
Merge branch 'dev' into tauquir/enclave-prototype

BrendanSchell committed:
first pass at doing migrations

BrendanSchell committed:
fix usercode upgrade to v7

rasswanth-s committed:
split notebooks into V2 for next iteration
add notes for project requirements

rasswanth-s committed:
re-organize file structure of network service

rasswanth-s committed:
added inter domain communication initiated by the Data Scientist

rasswanth-s committed:
Re-organization of folder structures

rasswanth-s committed:
renamed ping to challege nonce

rasswanth-s committed:
added ping endpoint to ping peers by verify key
add route checkability function for clients
fix Decentralized project submission

rasswanth-s committed:
added a single notebook to enclaves for V2 iteration for faster development
add nbstrip to enclave notebooks

rasswanth-s committed:
add SubmitUserCode as event to Project

rasswanth-s committed:
added automatic request creation to project

rasswanth-s committed:
added post hook events for request response events

rasswanth-s committed:
Added stub for project code status

itstauq committed:
Merge branch 'dev' into tauquir/enclave-prototype

rasswanth-s committed:
added code status feature based on projects as single source of truth

rasswanth-s committed:
added new code and requests interface to projects
modify print to print the NodeIdentity  for easy identification of code requests

rasswanth-s committed:
added code status to the e2e flow notebooks

madhavajay committed:
Merge branch 'bschell/rename-deployment-policy-to-runtime-policy' into tauquir/enclave-prototype

madhavajay committed:
Merge branch 'rasswanth/switch-to-old-projects' into tauquir/enclave-prototype

shubham3121 committed:
only keep user code version 5, merge all other changes into version 5
- rebuild protocol version

shubham3121 committed:
make usercodev4 serialized and restage protocol version

shubham3121 committed:
Merge branch 'tauquir/enclave-prototype' into bschell/update-migrations-for-enclave-pr

shubham3121 committed:
Merge branch 'dev' into tauquir/enclave-prototype

shubham3121 committed:
Merge branch 'tauquir/enclave-prototype' into bschell/update-migrations-for-enclave-pr

shubham3121 committed:
rename deployment prefix to runtime in submit usercode

shubham3121 committed:
Merge pull request #8986 from OpenMined/bschell/update-migrations-for-enclave-pr

Add Migrations for Enclave Changes to UserCode and CodeHistory

madhavajay committed:
Added notebook

- added SyftModel prototype
- pulled in Beach Model code
- added SyftFolder / SyftFile abstraction

itstauq committed:
Add methods for manual orchestration by the DS to the ProjectCode class

rasswanth-s committed:
added code to write model folder to current path
shifted to new model hosting notebook for testing

itstauq committed:
Merge branch 'dev' into tauquir/enclave-prototype

rasswanth-s committed:
added notebooks for training a simple model

itstauq committed:
Merge branch 'tauquir/enclave-prototype' into tauquir/enclave-manual-orchestration

itstauq committed:
Fix linters

itstauq committed:
Fix action objects not getting saved in the database

itstauq committed:
Merge branch 'dev' into tauquir/enclave-prototype

itstauq committed:
Add get_result and orchestrate_enclave_execution methods to ProjectCode

itstauq committed:
Merge branch 'tauquir/enclave-prototype' into tauquir/enclave-manual-orchestration

itstauq committed:
Remove unnecessary arguments from the action_service.set method

itstauq committed:
Add manual orchestration steps in 04-ds-execute-code notebook

rasswanth-s committed:
remove unused model_code in model file

rasswanth-s committed:
Merge branch 'dev' into madhava/model_hosting_prototype

rasswanth-s committed:
finished model hosting iteration using datasets until architecture for model hosting is freezed

rasswanth-s committed:
add endpoint ideas notebook

itstauq committed:
Merge pull request #9020 from OpenMined/tauquir/enclave-manual-orchestration

Enclave manual orchestration for Data Scientists

itstauq committed:
Merge branch 'dev' into tauquir/enclave-prototype

itstauq committed:
Make Enclave prototype work with Kubernetes cluster

yashgorana committed:
update notebook

yashgorana committed:
fix variable names

madhavajay committed:
Merge pull request #9029 from OpenMined/tauquir/enclave-k8s-support

Make Enclave prototype work with Kubernetes cluster

madhavajay committed:
Updated enclave notebooks

- Made changes to notification code to fix bugs
- Added get_nb_secrets util

madhavajay committed:
Added code to capture model class in decorator

madhavajay committed:
Merge branch 'madhava/model_hosting_prototype' into tauquir/enclave-prototype

madhavajay committed:
Added model upload to first notebook

- No mock yet, just duplicate model data

rasswanth-s committed:
added execution context to Submit Code

rasswanth-s committed:
Merge branch 'dev' into tauquir/enclave-prototype

rasswanth-s commented:
Merging in DEV, as I require the skip_save_to_blob_store functionality from dev.

rasswanth-s committed:
Merge branch 'tauquir/enclave-prototype' into rasswanth/model_hosting_v2

rasswanth-s committed:
added model ref action object to dynamically load models

rasswanth-s committed:
add ModelRef as a ProxyActionObject to load model
add ModelRef to upload dataset
fixes model inference on eval dataset

itstauq committed:
Make Enclave V2 notebook work with remote enclaves

itstauq committed:
Add get_enclave_attestation method to ProjectCode class

itstauq committed:
Get raw attestation token from enclave

rasswanth-s committed:
Merge branch 'tauquir/enclave-prototype' into rasswanth/model_hosting_v2

rasswanth-s committed:
Merge branch 'dev' into tauquir/enclave-prototype

rasswanth-s committed:
Merge branch 'tauquir/enclave-prototype' into rasswanth/model_hosting_v2

rasswanth-s committed:
modify submit model code to action object

rasswanth-s committed:
Shifted to Nested Paradigm of Model Initialization in UserCode
modify submit_model to code_action_id

rasswanth-s committed:
Added combine assets transfer for enclaves
Add utility functions to wrap and unwrap initialized model
Created a new notebook, for model-hosting + enclaves

itstauq committed:
Add view_attestation_report method to ProjectCode class

itstauq committed:
Add a TODO to hardcode the JWKS URL to prevent vulnerability

itstauq committed:
Add GPU Enclave attestation support

rasswanth-s committed:
update model hosting notebooks

itstauq committed:
Add test-gpu Enclave endpoint

itstauq committed:
Update notebook to use CPU enclave instead of GPU

rasswanth-s committed:
upload to enclave blob storage before transferring assets to enclave
add client argument to save to blob storage

rasswanth-s committed:
cleanup test notebooks format

itstauq committed:
Merge pull request #9036 from OpenMined/tauquir/enclave-attestations

View Enclave attestation report

itstauq committed:
CI fixes - Part 1

itstauq committed:
Merge pull request #9031 from OpenMined/rasswanth/model_hosting_v2

Model Hosting Revamp

itstauq committed:
Fix linters

itstauq committed:
Comment out print statement with sensitive information

itstauq committed:
Fix CI

rasswanth-s committed:
add gpu support to k3d nodes
created new devspace profiles for cpu and gpu enclaves
add readme to setup enclave for k3d
add env parameter to launch gpu enclave easily

rasswanth-s committed:
fix format error

rasswanth-s committed:
add k3s to repo args

rasswanth-s committed:
added gpu support with devspace profiles

rasswanth-s committed:
fix file name of gpu values.yaml

itstauq committed:
Optimization 1

itstauq committed:
Remove asset_list from model before saving it to the model stash

itstauq committed:
Attestation reports revamp

rasswanth-s committed:
upload digital signature benchmarks

rasswanth-s committed:
upload digital signature benchmarks for ecc, rsa

rasswanth-s committed:
Add benchmarks also for pycryptodome

rasswanth-s committed:
clear up and data and mock references in asset before uploading to server , as they are referenced by id's

rasswanth-s committed:
clear up action data cache when uploading to blob store

rasswanth-s committed:
add pyinstrument profiling

rasswanth-s committed:
add basic hash calculation function to assets

rasswanth-s committed:
add basic model hash functions

rasswanth-s committed:
add notebooks for hash testing

rasswanth-s committed:
cleanup profile settings

rasswanth-s committed:
change default profile interval to be 0.001

rasswanth-s committed:
revert to sync routes

rasswanth-s committed:
shift pyinstrument to make_routes, as async routes are not fully supported,
disable pyinstrument middlware temporarily
re-name appsettings to serversettings

rasswanth-s committed:
Merge pull request #9053 from OpenMined/tauquir/enclave-model-upload-optimization

Enclave model upload optimization

rasswanth-s committed:
Merge branch 'tauquir/enclave-prototype' into rasswanth/structured-transparency

rasswanth-s committed:
Merge branch 'dev' into tauquir/enclave-prototype

rasswanth-s committed:
Merge branch 'dev' into tauquir/enclave-prototype

rasswanth-s committed:
Merge branch 'dev' into tauquir/enclave-prototype

rasswanth-s committed:
Merge branch 'dev' into tauquir/enclave-prototype

rasswanth-s committed:
lint and merge with enclave branch

rasswanth-s committed:
stage protocol version for asset class

rasswanth-s committed:
fix left out re-name errors

rasswanth-s committed:
added model hashing to model ref action object

rasswanth-s committed:
added hash calculation at action service level

rasswanth-s committed:
add hash information to user code and submit user code

rasswanth-s committed:
add temporary performance speedup for in-memory workers

rasswanth-s committed:
fix pydantic reserved names in model.py
action action object benchmark suite

rasswanth-s committed:
fix caching and blob storage intialization

rasswanth-s committed:
fix model ref retrieval

rasswanth-s committed:
update notebook

rasswanth-s committed:
temporarily remove performance optimization
add inputid2hash to submit user code
remove old enclave.code

rasswanth-s committed:
add input verification and output verification to enclave result

rasswanth-s committed:
Merge branch 'tauquir/enclave-prototype' into rasswanth/gpu-k3d

rasswanth-s committed:
Merge pull request #9051 from OpenMined/rasswanth/gpu-k3d

Add GPU Support to k3d

rasswanth-s committed:
Merge branch 'tauquir/enclave-prototype' into rasswanth/structured-transparency

rasswanth-s committed:
Merge pull request #9073 from OpenMined/rasswanth/structured-transparency

[WIP] Structured Transparency

rasswanth-s committed:
update demo notebook

rasswanth-s committed:
add host 0.0.0.0 for container host

rasswanth-s committed:
update notebook

rasswanth-s committed:
Merge pull request #9097 from OpenMined/rasswanth/k8s-enclave

Fix Enclave Bugs in K8s Testing

rasswanth-s committed:
added script to reset kubernetes resources for faster development

rasswanth-s committed:
parallelize storage deletion

rasswanth-s committed:
update notebooks with k8s instructions for reset

rasswanth-s committed:
Merge pull request #9098 from OpenMined/rasswanth/fix-k8s-local

Fix Local K8s for Enclave

rasswanth-s committed:
modified v3 notebooks to be generic for
1. in-memory workers
2.Local K8s
3. remote k8s

rasswanth-s committed:
fix blob storage errors

itstauq committed:
Add ability to directly specify enclave route as a string

rasswanth-s committed:
Add Mock Model Execution

itstauq committed:
Add util functions to exchange peer routes and check reachability

itstauq committed:
Add default port for the route if not provided

rasswanth-s committed:
add hugging face hub to pypi

rasswanth-s committed:
Shift from tqdm to rich

rasswanth-s committed:
Revert "Shift from tqdm to rich"

This reverts commit f44c200a9f9476392231a532678b3c24c17354b5.

rasswanth-s committed:
modified tqdm to print in-line

rasswanth-s committed:
Merge pull request #9126 from OpenMined/rasswanth/add-mock-execution

Add Mock Model Execution

rasswanth-s committed:
Merge branch 'tauquir/enclave-prototype' into tauquir/exchange-peer-routes

rasswanth-s committed:
add rich table for exchange routes

rasswanth-s committed:
add updated notebooks with Data Scientist requesting for route exchange

rasswanth-s committed:
Merge pull request #9127 from OpenMined/tauquir/exchange-peer-routes

Add util functions to exchange peer routes and check reachability

rasswanth-s committed:
Merge branch 'tauquir/enclave-prototype' into tauquir/simplify-adding-enclave-route

rasswanth-s committed:
update route addition, based on url

rasswanth-s committed:
Merge pull request #9124 from OpenMined/tauquir/simplify-adding-enclave-route

Add ability to directly specify enclave route as a string

itstauq committed:
Show pretty repr of VerifiableOutput

itstauq committed:
Improve enclave repr

itstauq committed:
Round the model size repr to two decimal places

teo-milea committed:
repr for model hosting objects

teo-milea committed:
fix lint

rasswanth-s committed:
Merge branch 'tauquir/enclave-prototype' into model_hosting_repr

teo-milea committed:
fix notebook

rasswanth-s committed:
update notebooks
add syft error for python attestation

rasswanth-s committed:
Merge pull request #9135 from OpenMined/model_hosting_repr

repr for model hosting objects

rasswanth-s committed:
add torch and transformers to the notebook

rasswanth-s committed:
Merge branch 'tauquir/enclave-prototype' into tauquir/pretty-verifiable-output-repr

rasswanth-s committed:
add datasite to repr

rasswanth-s committed:
Merge pull request #9131 from OpenMined/tauquir/pretty-verifiable-output-repr

Show pretty representation of VerifiableOutput

rasswanth-s committed:
Merge branch 'tauquir/enclave-prototype' into tauquir/improve-enclave-repr

rasswanth-s committed:
update V3 notebook showing enclave instance

rasswanth-s committed:
Merge pull request #9132 from OpenMined/tauquir/improve-enclave-repr

Improve enclave representation

rasswanth-s committed:
Merge branch 'tauquir/enclave-prototype' into tauquir/fix-model-size-repr

rasswanth-s committed:
fix Size Calculation of Models
Fix repr of Model with MB
Update V3 noteboooks to show model coll repr

rasswanth-s committed:
Merge pull request #9133 from OpenMined/tauquir/fix-model-size-repr

Fix model size calculation and improve repr

rasswanth-s committed:
Merge branch 'tauquir/enclave-prototype' into tauquir/attestation-summary

rasswanth-s committed:
add mock cpu and gpu report

tcp committed:
fix: look of syft responses unaffected by MarkdownDescription

In MarkdownDescription, a conflicting CSS style is added globally. I
have removed the !important tag, which will now make the other pre look
as they should; md descriptions are probably affected but I'm ignoring
it for now.

tcp committed:
fix: safeguard for unknown model size in repr

tcp committed:
fix: remove padding from MarkdownDescription

tcp committed:
feat: repr for exchange_routes

tcp committed:
fix: remove style override from MarkdownDescription

itstauq committed:
Add init, run and stop policy conditions to RunOnEnclave runtime policy

teo-milea committed:
autogenerate mock

teo-milea committed:
improve ux

teo-milea committed:
protocol version

teo-milea committed:
new notebook

rasswanth-s committed:
Add Events for
ProjectAssetTransfer
ProjectEnclaveAttestationReport
ProjectExeuctionStart
ProjectEnclaveReport Ready

rasswanth-s committed:
Merge pull request #9152 from OpenMined/rasswanth/add-project-events

Add Remaining Project Events

rasswanth-s committed:
Merge branch 'tauquir/enclave-prototype' into ux_model_hosting

tcp committed:
fix: SyftResponse str tag from pre to span

tcp committed:
feat: new styles for code blocks

tcp committed:
fix: Model* uses model_validator instead of init

tcp committed:
feat: html for ProjectCode

tcp committed:
fix: guard repr on ProjectRequest

itstauq committed:
Add serializable decorator and fix protocol_version.json

itstauq committed:
Merge branch 'tauquir/enclave-prototype' into tauquir/add-policy-conditions

rasswanth-s committed:
add mock model generation as a step in the notebook

rasswanth-s committed:
add auto generate at Model Intialization level
update V3 notebooks with latest interface
fixed eos token id error
remove auto generation on at service level

rasswanth-s committed:
delete new experimental notebook

rasswanth-s committed:
fix model mock assets auto generation

rasswanth-s committed:
update notebook with autogenerate instructions

rasswanth-s committed:
remove print statements in model.py
Tested in Local Kubernetes Cluster

rasswanth-s committed:
Merge pull request #9149 from OpenMined/ux_model_hosting

Ux model hosting

rasswanth-s committed:
Merge branch 'tauquir/enclave-prototype' into tauquir/add-policy-conditions

rasswanth-s committed:
Modify name PolicyCondition to RuntimePolicyCondition

rasswanth-s committed:
Merge pull request #9144 from OpenMined/tauquir/add-policy-conditions

Add init, run and stop policy conditions to RunOnEnclave runtime policy

rasswanth-s committed:
Merge branch 'tauquir/enclave-prototype' into tauquir/attestation-summary

rasswanth-s committed:
Install DS packages in syft Client

rasswanth-s committed:
Merge pull request #9153 from OpenMined/rasswanth/install-ds-packages-syft-client

[Enclave] install ds packages syft client

rasswanth-s committed:
Merge branch 'tauquir/enclave-prototype' into tauquir/attestation-summary

rasswanth-s committed:
Add GPU Attestation Report

rasswanth-s committed:
Add flag to verify report to ignore expiry for mock reports

tcp committed:
lint

rasswanth-s committed:
add mock attestation feature

rasswanth-s committed:
Merge pull request #9056 from OpenMined/tauquir/attestation-summary

Attestation reports revamp

tcp committed:
Merge remote-tracking branch 'origin/tauquir/enclave-prototype' into thiago/enclave-ux

tcp committed:
merge and lint

tcp committed:
fix: fstring err in <3.12

rasswanth-s committed:
Merge pull request #9142 from OpenMined/thiago/enclave-ux

feat: ux improvements to enclave reprs

rasswanth-s committed:
add asset name to project asset transfer event

rasswanth-s committed:
Merge branch 'tauquir/enclave-prototype' into rasswanth/add-asset-name

rasswanth-s committed:
Merge pull request #9158 from OpenMined/rasswanth/add-asset-name

Add Asset Name to Asset Transfer Event in Project

rasswanth-s committed:
add accelerate and inspect-ai to setup.cfg

rasswanth-s committed:
Merge pull request #9159 from OpenMined/rasswanth/add-additional-deps

Add Additional Deps to Enclave

itstauq committed:
Add get_code_status method to UserCode class

rasswanth-s committed:
Update scenario notebooks

rasswanth-s committed:
Merge branch 'tauquir/enclave-prototype' into tauquir/multiparty-usercode-status-repr

rasswanth-s committed:
Merge pull request #9161 from OpenMined/tauquir/multiparty-usercode-status-repr

Fix UserCode.status property and UserCode repr for multiparty Enclave codes

rasswanth-s committed:
add columns repr to project.events

rasswanth-s committed:
add code hash to project code

rasswanth-s committed:
Increased seaweedfs timeout for uploads

rasswanth-s committed:
Merge pull request #9164 from OpenMined/rasswanth/increase-timeout

Increased seaweedfs timeout for uploads

tcp committed:
feat: more ux improvements to events

rasswanth-s committed:
Merge pull request #9162 from OpenMined/rasswanth/add_coll_repr

Add Columns Repr to Project Events

itstauq committed:
Fix typo in tox.ini

rasswanth-s committed:
fix resource restart problem attesation

rasswanth-s committed:
fix gpu mock attestation report

rasswanth-s commented:
Hello 👋 

Thanks so much for this PR — I really appreciate the work you put into it.

A quick heads-up: the repo just went through a significant refactor, so the codebase has changed quite a bit from what this PR was built on. As a result, these changes no longer apply cleanly to the new structure, and I'm going to close this PR for now.

if you're interested in contributing to the current version, kindly have a look at the [README](https://github.com/OpenMined/PySyft/blob/dev/README.md)

Thanks again for contributing — and sorry for the churn!
